python/ray/data/datasource/file_based_datasource.py

```diff
@@ -132,6 +219,8 @@ def prepare_read(
         filesystem: Optional["pyarrow.fs.FileSystem"] = None,
         schema: Optional[Union[type, "pyarrow.lib.Schema"]] = None,
         open_stream_args: Optional[Dict[str, Any]] = None,
+        meta_provider: Optional[BlockMetadataProvider] = DefaultBlockMetadataProvider(),
```

Comment:
Yeah, I think I like that better given that the core interface always requires file paths to be specified, and it could help differentiate it from any non-file-based BlockMetadata providers that may come along later.